server/src/main/java/org/elasticsearch/search/SearchHit.java

```diff
@@ -449,7 +440,7 @@ public String getSourceAsString() {
      * The hit field matching the given field name.
      */
     public DocumentField field(String fieldName) {
-        return getFields().get(fieldName);
+        return documentFields.getOrDefault(fieldName, metaFields.get(fieldName));
```

Comment:
nit: can you use two separate lookups (look up in metaFields if the first fails)? I had a bit of trouble reading this.